src/app/Fake.DotNet.Testing.NUnit/NUnit3.fs

```diff
@@ -309,35 +309,50 @@ let buildArgs (parameters:NUnit3Params) (assemblies: string seq) =
     |> StringBuilder.appendFileNamesIfNotNull assemblies
     |> StringBuilder.toText
 
-let run (setParams : NUnit3Params -> NUnit3Params) (assemblies : string seq) =
-    let details = assemblies |> String.separated ", "
-    use __ = Trace.traceTask "NUnit" details
+let internal createProcess (setParams : NUnit3Params -> NUnit3Params) (assemblies : string[]) =
     let parameters = NUnit3Defaults |> setParams
-    let assemblies = assemblies |> Seq.toArray
     if Array.isEmpty assemblies then failwith "NUnit: cannot run tests (the assembly list is empty)."
     let tool = parameters.ToolPath
-    let args = buildArgs parameters assemblies
-    Trace.trace (tool + " " + args)
-    let processTimeout = TimeSpan.MaxValue // Don't set a process timeout. The timeout is per test.
-    let result =
-        Process.execSimple ((fun info ->
-        { info with
-            FileName = tool
-            WorkingDirectory = getWorkingDir parameters
-            Arguments = args }) >> Process.withFramework) processTimeout
-    let errorDescription error =
-        match error with
-        | OK -> "OK"
-        | TestsFailed -> sprintf "NUnit test failed (%d)." error
-        | FatalError x -> sprintf "NUnit test failed. Process finished with exit code %s (%d)." x error
-
-    match parameters.ErrorLevel with
-    | NUnit3ErrorLevel.DontFailBuild ->
-        match result with
-        | OK | TestsFailed -> ()
-        | _ -> raise (FailedTestsException(errorDescription result))
-    | NUnit3ErrorLevel.Error | FailOnFirstError ->
-        match result with
-        | OK -> ()
-        | _ -> raise (FailedTestsException(errorDescription result))
-    __.MarkSuccess()
+    let generatedArgs = buildArgs parameters assemblies
+    //let processTimeout = TimeSpan.MaxValue // Don't set a process timeout. The timeout is per test.
+    
+    let path = Path.GetTempFileName()
+    let args = (sprintf "@%s" path)
+    CreateProcess.fromRawWindowsCommandLine tool args
+    |> CreateProcess.withFramework
+    |> CreateProcess.withWorkingDirectory (getWorkingDir parameters)
+    //|> CreateProcess.withTimeout processTimeout
```

Comment:
I think we want this max timeout to ensure that the process doesn;t get shutdown.
Nunit has it's own timeout built in as an argument.

Reply:
The `CreateProcess` has no default timeout, therefore it waits forever by default which seems like the intended behavior.

Reply:
Fair enough!